src/domain/event/hooks/useUpdateRecurringEventIfNeeded.tsx

```diff
@@ -0,0 +1,141 @@
+import { useApolloClient } from '@apollo/client';
+import isEqual from 'date-fns/isEqual';
+import isNull from 'lodash/isNull';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { useLocation } from 'react-router';
+
+import {
+  EventFieldsFragment,
+  PublicationStatus,
+  SuperEventType,
+  useUpdateEventMutation,
+} from '../../../generated/graphql';
+import parseIdFromAtId from '../../../utils/parseIdFromAtId';
+import { reportError } from '../../app/sentry/utils';
+import { authenticatedSelector } from '../../auth/selectors';
+import useUser from '../../user/hooks/useUser';
+import { EVENT_EDIT_ACTIONS } from '../constants';
+import { EventTime } from '../types';
+import {
+  calculateSuperEventTime,
+  checkIsEditActionAllowed,
+  getEventInitialValues,
+  getEventPayload,
+  getOrganizationAncestors,
+  getRecurringEvent,
+} from '../utils';
+
+type UpdateRecurringEventIfNeededState = {
+  updateRecurringEventIfNeeded: (values: EventFieldsFragment) => void;
+};
+
+const useUpdateRecurringEventIfNeeded = (): UpdateRecurringEventIfNeededState => {
+  const apolloClient = useApolloClient();
+  const location = useLocation();
+  const { t } = useTranslation();
+  const authenticated = useSelector(authenticatedSelector);
+  const { user } = useUser();
+  const [updateEvent] = useUpdateEventMutation();
+
+  const shouldUpdateTime = (time: Date | null, newTime: Date | null) =>
+    ((isNull(time) || isNull(newTime)) && time !== newTime) ||
+    (time && newTime && !isEqual(time, newTime));
+
+  const updateRecurringEventIfNeeded = async (event: EventFieldsFragment) => {
```

Comment:
Some explanation about this could be helpful.

Reply:
Added a comment